core/kazoo_tasks/src/kz_csv.erl

```diff
@@ -90,32 +95,104 @@ fold(CSV, Fun, Acc)
+split_field(Row, EndChar, Fields) ->
+    split_field(Row, EndChar, Fields, []).
+
+split_field(<<$,>>, $,, Fields, FieldSoFar) ->
+    split_fields(<<>>, [<<>>, iolist_to_binary(lists:reverse(FieldSoFar)) | Fields]);
+split_field(<<$,, Row/binary>>, $,, Fields, FieldSoFar) ->
+    split_fields(Row, [iolist_to_binary(lists:reverse(FieldSoFar)) | Fields]);
+split_field(<<"\n\r", _/binary>>=Row, $,, Fields, FieldSoFar) ->
+    split_fields(Row, [iolist_to_binary(lists:reverse(FieldSoFar)) | Fields]);
+split_field(<<"\r\n", _/binary>>=Row, $,, Fields, FieldSoFar) ->
+    split_fields(Row, [iolist_to_binary(lists:reverse(FieldSoFar)) | Fields]);
+split_field(<<"\r\r", _/binary>>=Row, $,, Fields, FieldSoFar) ->
+    split_fields(Row, [iolist_to_binary(lists:reverse(FieldSoFar)) | Fields]);
+split_field(<<"\n", _/binary>>=Row, $,, Fields, FieldSoFar) ->
+    split_fields(Row, [iolist_to_binary(lists:reverse(FieldSoFar)) | Fields]);
+split_field(<<"\r", _/binary>>=Row, $,, Fields, FieldSoFar) ->
+    split_fields(Row, [iolist_to_binary(lists:reverse(FieldSoFar)) | Fields]);
+
+split_field(<<EndChar, EndChar, Row/binary>>, EndChar, Fields, FieldSoFar) ->
+    split_field(Row, EndChar, Fields, [EndChar, EndChar | FieldSoFar]);
+split_field(<<EndChar, $,, Row/binary>>, EndChar, Fields, FieldSoFar) ->
+    Field = iolist_to_binary(lists:reverse(FieldSoFar)),
+    split_fields(Row, [Field | Fields]);
+
+split_field(<<EndChar, "\r\n", Row/binary>>, EndChar, Fields, FieldSoFar) ->
+    Field = iolist_to_binary(lists:reverse(FieldSoFar)),
+    split_fields(<<"\r\n", Row/binary>>, [Field | Fields]);
+split_field(<<EndChar, "\n", Row/binary>>, EndChar, Fields, FieldSoFar) ->
+    Field = iolist_to_binary(lists:reverse(FieldSoFar)),
+    split_fields(<<"\n", Row/binary>>, [Field | Fields]);
+split_field(<<EndChar, "\r\r", Row/binary>>, EndChar, Fields, FieldSoFar) ->
+    Field = iolist_to_binary(lists:reverse(FieldSoFar)),
+    split_fields(<<"\r\r", Row/binary>>, [Field | Fields]);
+split_field(<<EndChar, "\r\n", Row/binary>>, EndChar, Fields, FieldSoFar) ->
+    Field = iolist_to_binary(lists:reverse(FieldSoFar)),
+    split_fields(<<"\r\n", Row/binary>>, [Field | Fields]);
+split_field(<<EndChar, "\r", Row/binary>>, EndChar, Fields, FieldSoFar) ->
+    Field = iolist_to_binary(lists:reverse(FieldSoFar)),
+    split_fields(<<"\r", Row/binary>>, [Field | Fields]);
+
+split_field(<<>>, $,, Fields, FieldSoFar) ->
+    split_fields(<<>>, [<<>>, iolist_to_binary(lists:reverse(FieldSoFar)) | Fields]);
+split_field(<<EndChar>>, EndChar, Fields, FieldSoFar) ->
+    Field = iolist_to_binary(lists:reverse(FieldSoFar)),
+    split_fields(<<>>, [Field | Fields]);
+
+split_field(<<Char:1/binary>>, _EndChar, Fields, FieldSoFar) ->
+    Field = iolist_to_binary(lists:reverse([Char | FieldSoFar])),
+    split_fields(<<>>, [Field | Fields]);
+split_field(<<Char:8, Row/binary>>, EndChar, Fields, FieldSoFar) ->
+    split_field(Row, EndChar, Fields, [Char | FieldSoFar]).
```

Comment:
I don't get it: why accumulate a binary into a reversed list of binaries when you can just `<<FieldSoFar/binary, Char/binary>>`?